src/main/java/org/opensearch/ad/feature/SearchFeatureDao.java

```diff
@@ -118,10 +116,10 @@ public SearchFeatureDao(
         this.maxEntitiesForPreview = maxEntitiesForPreview;
         this.pageSize = pageSize;
         try {
-            Map<Setting<?>, Consumer<?>> settingsUpdateConsumers = new HashMap<>();
-            settingsUpdateConsumers.put(MAX_ENTITIES_FOR_PREVIEW, it -> this.maxEntitiesForPreview = (int) it);
-            settingsUpdateConsumers.put(PAGE_SIZE, it -> this.pageSize = (int) it);
-            clusterService.getClusterSettings().addSettingsUpdateConsumer(settingsUpdateConsumers);
+            clusterService
+                .getClusterSettings()
+                .addSettingsUpdateConsumer(MAX_ENTITIES_FOR_PREVIEW, it -> this.maxEntitiesForPreview = (int) it);
```

Comment:
Two things:
1. You can remove the try/catch.  That was fixed in a previous SDK PR.
2. You don't need to cast anymore. That was needed for the map, but the other method has a generic type.

Best practice: look at this same class on the `main` branch, find the lines, and copy-paste them over the existing try/catch sequence.